dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java

```diff
@@ -66,11 +66,13 @@ public NDList forward(
         // TODO maybe verify the number of inputs
         // currently we assume the order of the input NDList is the same
         // as the model input
-        for (int i = 0; i < inputs.size(); ++i) {
-            JniUtils.setDlrInput(modelHandle, inputs.get(i), i);
+        synchronized (DlrSymbolBlock.class) {
```

Comment:
This is not the true multi-threading. More like a thread-safe forward method.

Reply:
right as it don't support real multithreading if we don't want to copy model & dlr lib